client/finality-grandpa/src/finality_proof.rs

```diff
@@ -338,10 +337,12 @@ pub(crate) fn prove_finality<Block: BlockT<Hash=H256>, B: BlockchainBackend<Bloc
 			if justifies_end_block || justifies_authority_set_change {
 				// check if the proof is generated by the requested authority set
 				if finality_proof.is_empty() {
```

Comment:
Why does it only check that the justification is correct if there are no other proof fragments?

Reply:
Because, as I said above, we only check that the **initial set** is the same on both requester and prover